packages/dds/tree/src/shared-tree-core/editManager.ts

```diff
@@ -93,14 +120,69 @@ export class EditManager<
 			revision: nullRevisionTag,
 			sessionId: "",
 			change: changeFamily.rebaser.compose([]),
+			sequenceNumber: minimumPossibleSequenceNumber,
 		};
 		this.trunk = this.trunkBase;
 		this.localBranch = this.trunk;
 	}
 
+	/**
+	 * Make the given branch known to the `EditManager`. The `EditManager` will ensure that all registered
+	 * branches remain usable even as the minimum sequence number advances.
+	 * @param branch - the branch to register. All branches that fork from this branch, directly or transitively,
+	 * will also be registered.
+	 */
+	public registerBranch(branch: SharedTreeBranch<ChangeFamilyEditor, TChangeset>): void {
```

Comment:
what happens if you double register?

Reply:
We talked offline about this too, but now that I think about it, double registering should work okay. The branch will bump the counter in the map twice instead of once, and the same will happen for all its forks. But, they will also subscribe to the dispose event twice, so that when they are disposed, the counter will decrement twice. So, the counter will be wrong when it isn't "0", but "0" is the only state we actually care about, so it works out. 

It still seems like it'd be worth it to disallow it, but in order to do that we'd have to actually keep around the references to the branch objects themselves in the map to detect collisions. Which I suppose means having the value in the map be a Set<Branch> rather than a counter. Bleh. Probably it will never matter, but it's probably still the right thing to do.